File: frontend/types/submission.ts
export type SubmissionStatus = "Submitted" | "Graded";

export interface Submission {
  id: number;
  assignmentId: number;
  assignmentTitle: string;
  studentId: number;
  studentName: string;
  content: string;
  submittedAt: string;
  updatedAt: string | null;
  status: SubmissionStatus;
  marks: number | null;
  feedback: string | null;
  maxMarks: number;
}

export interface CreateSubmissionInput {
  assignmentId: number;
  content: string;
}

export interface UpdateSubmissionInput {
  content: string;
}

export interface GradeSubmissionInput {
  marks: number;
  feedback: string;
}

File: backend/SchoolMS.Data/Entities/Submission.cs
using SchoolMS.Data.Enums;

namespace SchoolMS.Data.Entities;

public class Submission
{
    public int Id { get; set; }

    public int AssignmentId { get; set; }
    public Assignment Assignment { get; set; } = null!;

    public int StudentId { get; set; }
    public User Student { get; set; } = null!;

    public string Content { get; set; } = string.Empty;
    public DateTime SubmittedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public SubmissionStatus Status { get; set; } = SubmissionStatus.Submitted;
    public int? Marks { get; set; }
    public string? Feedback { get; set; }
}
